### PR TITLE
[CI] Use Mill for testing

### PR DESCRIPTION
**Pull request by jackkoenig**
https://github.com/chipsalliance/chisel/pull/4462 broke Chisel because I accidentally deleted the scalacOption for the plugin from the Chisel module. This was missed because CI is still run with SBT. This PR will fix the issue and switch to using Mill for CI so we don't make this mistake again.

### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Bugfix
- Internal or build-related (includes code refactoring/cleanup)

#### Desired Merge Strategy

- Squash

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by jackkoenig** [comment]
First testing the CI without the fix to confirm we can detect the previous breakage.

**Comment by jackkoenig** [comment]
I switched the Label to "Internal" because this is really a CI change, and the bug is only visible on 1 SNAPSHOT commit, no released versions.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -207,9 +209,8 @@ jobs:
         with:
           distribution: 'adopt'
           java-version: '11'
-          cache: 'sbt'
-      - name: Unit Tests
-        run: sbt ++${{ inputs.scala }} standardLibrary/test
+      - name: Compile
+        run: ./mill stdlib[_].compile
```
Comment (jackkoenig) [inline .github/workflows/test.yml:213]: There are no stdlib tests...

File: core/src/main/scala/chisel3/internal/Builder.scala [hunk 2]
```diff
@@ -813,13 +813,23 @@ private[chisel3] object Builder extends LazyLogging {
...
+        dc.value
+      } catch {
+        // If user uses -Xcheckinit there will be additional information
+        case UninitializedFieldError(msg) => err(msg)
+      }
+    // If user is not using -Xcheckinit, suggest that they do
+    if (result == null) {
+      err("Try adding -Xcheckinit to your scalacOptions to get a more useful stack trace")
+    }
     result
```
Comment (jackkoenig) [inline core/src/main/scala/chisel3/internal/Builder.scala:833]: This was needed to fix some tests that failed due to our now using `-Xcheckinit`. I made it such that the error suggestion works both with and without `-Xcheckinit` and either gives you the additional context from using that option or suggests you use `-Xcheckinit`.

File: core/src/main/scala/chisel3/internal/Builder.scala [hunk 3]
```diff
@@ -813,13 +813,23 @@ private[chisel3] object Builder extends LazyLogging {
...
+    val result =
+      try {
+        dc.value
+      } catch {
+        // If user uses -Xcheckinit there will be additional information
+        case UninitializedFieldError(msg) => err(msg)
+      }
+    // If user is not using -Xcheckinit, suggest that they do
+    if (result == null) {
+      err("Try adding -Xcheckinit to your scalacOptions to get a more useful stack trace")
```
Comment (seldridge) [inline core/src/main/scala/chisel3/internal/Builder.scala:831]: Nit: missing trailing period.

**Review by seldridge — APPROVED** [review]
LGTM

File: core/src/main/scala/chisel3/internal/Builder.scala (hunk 3 above)
Comment (jackkoenig) [reply core/src/main/scala/chisel3/internal/Builder.scala:831]: The trailing period is added by `def err` above! But had forgotten to add trailing periods to the comments

**Pull request merged**